trio/_util.py

```diff
@@ -338,7 +358,7 @@ def _create(cls: t.Type[T], *args: object, **kwargs: object) -> T:
         return super().__call__(*args, **kwargs)  # type: ignore
 
 
-def name_asyncgen(agen):
+def name_asyncgen(agen: collections.abc.AsyncGenerator[object, t.NoReturn]) -> str:
```

Comment:
In #2726 I ended up using `types.AsyncGeneratorType` here instead because it has the attributes this function uses defined unlike `collections.abc.AsyncGenerator`, and then you don't have errors that `__qualname__` doesn't exist